ui/src/main/webapp/wizards/ldap/stages/attribute-mapping.js

```diff
@@ -177,100 +184,127 @@ const testClaimMappings = (conn, info, loginUserAttribute, dn, mapping) => ({
-                  onEndSubmit()
-                  next('confirm')
-                })
-                .catch((err) => {
-                  onEndSubmit()
-                  onError(err)
-                })
-            }}
-            disabled={Object.keys(attributeMappings).length === 0} />
-        </Navigation>
-        {messages.map((msg, i) => <Message key={i} {...msg} />)}
-      </Body>
-    </div>
-  )
+    render() {
+        const {
+            client,
+            onError,
+            onStartSubmit,
+            onEndSubmit,
+            next,
+
+            configs,
+            configs: {
+                attributeMappings = {}
+            } = {},
+            onEdit,
+
+            errors,
+
+            data: { sts = {}, ldap = {} },
+
+            prev
+        } = this.props
+
+
+        // filter out unique messages
+        const messages = errors.filter((err, i) => {
+            const before = errors.slice(0, i).map((err) => err.message)
+            return !before.includes(err.message)
+        })
+
+        const conn = {
+            hostname: configs.hostname,
+            port: configs.port,
+            encryption: configs.encryption
+        }
+
+        const info = {
+            creds: {
+                username: configs.bindUser,
+                password: configs.bindUserPassword
+            },
+            bindMethod: configs.bindUserMethod,
+            realm: configs.bindRealm
+        }
+
+        const loginUserAttribute = configs.loginUserAttribute
+        const dn = configs.baseUserDn
+        const mapping = Object.keys(attributeMappings).map((key) => ({ key, value: attributeMappings[key] }))
+
+        return (
+            <div>
+              <Title>LDAP User Attribute Mapping</Title>
+              <Description>
+                In order to authorize users, their attributes must be mapped to the Security Token
+                Service (STS) claims. Not all attributes must be mapped but any unmapped attributes
+                will not be used for authorization.
+              </Description>
+
+              <AttributeMapper
+                  errors={this.state.mappedErrors}
+                  setMappedErrors={this.setMappedErrors}
+                  onEdit={onEdit}
+                  claims={sts.claims}
+                  attributes={ldap.userAttributes}
+                  configs={configs} />
+
+              <Body>
+              <Navigation>
+                <Back onClick={prev} />
+                <Next
+                    onClick={() => {
+                        onStartSubmit()
+                        client.query(testClaimMappings(conn, info, loginUserAttribute, dn, mapping))
+                            .then(() => {
+                                onEndSubmit()
+                                next('confirm')
+                            })
+                            .catch((err) => {
+                                onEndSubmit()
+                                onError(err)
```

Comment:
:question: We only need to do this in the catch?

Reply:
correct. on success the `next` action will cause all the errors to be reset.